project/src/main/scala/docs/JsonEncoder.scala

```diff
@@ -0,0 +1,37 @@
+package docs
+
+import better.files.Dsl.SymbolicOperations
+import better.files._
+import docs.transformers._
+import docs.model._
+
+object JsonEncoder {
+
+  /** Used to generate the
+    * "resources/docs/patterns.json" file.
+    *
+    * Gets a list of Patterns, returns the encoded json string.
+    */
+  def patternsJsonEncoder(version: String, patterns: Seq[Pattern]): String = {
+    import io.circe._, io.circe.generic.auto._, io.circe.syntax._
+    case class PatternFile(name: String,
+                           version: String,
+                           patterns: Seq[Pattern])
+    implicit val encodePattern: Encoder[Pattern] =
```

Comment:
If you import the `io.circe.generic.auto._` everything related to encoder should be done for free.